dialog/base.go

```diff
@@ -260,7 +268,7 @@ func (l *dialogLayout) Layout(obj []fyne.CanvasObject, size fyne.Size) {
 	l.d.bg.Move(fyne.NewPos(0, 0))
 	l.d.bg.Resize(size)
 
-	btnMin := obj[3].MinSize().Max(obj[3].Size())
+	btnMin := obj[3].MinSize()
```

Comment:
I have removed the part `.Max(obj[3].Size())` because it makes the dialog button ignore the theme change (sizing variables). @andydotxyz do you know if there was a reason to have that part? 
By removing it, a test image from the color picker changed, however it seems that before this change, that test image was wrong:
Left side is after this change and right side is before this change:

<img width="946" alt="Captura de Pantalla 2021-02-16 a la(s) 01 36 49" src="https://user-images.githubusercontent.com/12239342/108029136-f5b94900-6ffa-11eb-9329-a782a4d71d39.png">

Reply:
I don't recall - but this is the change https://github.com/fyne-io/fyne/commit/72cac28414e59cd6d6dd09fddfeb52caade7c4d2

Reply:
Hmm I see, that change breaks `dialog.NewProgress...`, however:
`btnMin := obj[3].MinSize().Max(obj[3].Size())`
Isn't the above code breaking the principle that objects inside a Layout cannot manipulate manually their size neither their position? @andydotxyz

Reply:
Also, I have seen that both `dialog.NewProgress` and `dialog.NewProgressInfinite` are deprecated and the message says:
```go
// Deprecated: Create a new custom dialog with a widget.ProgressBarInfinite() inside.
```
So using `dialog.NewCustom` the widget would be inside dialog content and not as a button as it is done currently inside `dialog.NewProgress`. 
However it is still possible to make the dialog.NewProgress looks like before, doing this:
```go
c := container.NewVBox(widget.NewProgressBarInfinite())
d := dialog.NewCustom("Title", "", c, w)
d.Resize(fyne.NewSize(200, d.MinSize().Height))
d.Show()
```
But NewCustom always render a dismiss button, I think there should be a way to tell dialog.NewCustom to not render that button if user does not want it.

Reply:
I think you are describing `widget.NewModalPopUp`

Reply:
Sorry, I didn't understand, what do you mean?

Reply:
Sorry, what I was meaning is: isn't a dialog with no buttons a modal popup?
I.e. dialogs are dismissable using buttons, to do something that isn't you can show a modal popup.

Reply:
Either way, for now perhaps the right thing to do is place a comment next to that line saying "// TODO remove once progress dialogs are gone" - if there is no sensible way to fix the issue with this PR.

Reply:
Maybe the comment in the deprecation should be fixed.
Anyhow, as I am sure you agree, we can't break the feature even if it's deprecated.

Reply:
> dialogs are dismissable using buttons, to do something that isn't you can show a modal popup.

Oh I see, yes you are right.

> for now perhaps the right thing to do is place a comment next to that line saying "// TODO remove once progress dialogs are gone"

By doing it, a wrong test image needs to be committed, is that ok?

> if there is no sensible way to fix the issue with this PR.

We could update dialog.NewProgress with the above code, then we don't need to commit wrong test images, while ensuring the same behavior.

> Maybe the comment in the deprecation should be fixed.

Yes, I think.

> Anyhow, as I am sure you agree, we can't break the feature even if it's deprecated.

Completely, agreed.